### PR TITLE
[CR] Add VSX Hadamard

### PR DESCRIPTION
**Pull request by rafaeldelucena**
## Description

This PR add Hadamard Transform using the VSX instructions set, aiming to optimize this operation for Power8 in order to resolve the issue #6

## Checklist:

- [x] Add code to Build System
- [x] Implement the Transpose 8x8 for signed shorts
- [x] Add Unit tests
- [x] Implement Hadamard 8x8
- [x] Implement Hadamard 16x16
- [x] Test for endianness and unaligned addresses issues
- [x] Make all tests pass

### CODE REVIEW

**Comment by rafaeldelucena** [comment]
@lu-zero
I've successful test the implementation on both little and big endian architectures, also I'm using the `vec_vsx_ld` and `vec_vsx_st` to load and store data, so I'll mark the `Test for endianness and unaligned addresses issues` as checked for now, I think it's ready for review. :)

**Comment by rafaeldelucena** [comment]
This is a first version, I don't apply any further optimizations, but if you find it necessary I can make the `vpx_transpose_8x8_s16` and `vpx_hadamard_8x8_s16_one_pass` inline functions also I can use an aligned address buffer and replace to simple load and store functions for char arrays, aiming to reduce the permutation overhead of the used instructions.

File: vpx_dsp/ppc/hadamard_vsx.c [hunk 1]
```diff
@@ -0,0 +1,153 @@
+#include <altivec.h>
+#include "vpx_dsp_rtcd.h"
+
+static void vpx_transpose_8x8_s16(vector signed short *a0,
```
Comment (lu-zero) [inline vpx_dsp/ppc/hadamard_vsx.c]: Benchmark if adding `inline` or `always_inline` improves the performance.

**Comment by lu-zero** [comment]
Using the test harness I'm not seeing any speedup, I guess some more work might be needed.

**Comment by lu-zero** [comment]
Seems that the test harness does not double as benchmark harness easily, do you have the alternative benchmark code ready to be in the tree?

**Comment by rafaeldelucena** [comment]
I've some test code here, I'll push until the end of the day.

**Comment by rafaeldelucena** [comment]
The benchmark code is very simple, runs the hadamard implementation with multiple strides from 10^1 to 10^6 times with the same input matrix.

**Comment by rafaeldelucena** [comment]
Already merged to upstream

**Pull request closed without merging**